RecoTracker/TkSeedGenerator/plugins/SeedingOTEDProducer.cc

```diff
@@ -0,0 +1,392 @@
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTracker.h"
+#include "TrackingTools/MeasurementDet/interface/LayerMeasurements.h"
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitMomentumHelper.h"
+
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTracker.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTrackerEvent.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/PatternTools/interface/TrajectoryStateUpdator.h"
+
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "FWCore/Utilities/interface/ESGetToken.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/MeasurementDet/interface/TrajectoryMeasurementGroup.h"
+
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class TrajectoryStateUpdator;
+
+class SeedingOTEDProducer : public edm::stream::EDProducer<> {
```

Comment:
I suspect this producer could be a `global`, at the expense of making some objects like `updator_` local and then passing them to the helper functions. The memory usage of this producer as a `stream` module should be assessed.

Reply:
For my education, what would be the advantage of making it `global`?

Reply:
A `stream` module is actually replicated multiple times, 1 time for each concurrent Event the job will run. There is only 1 copy of a `global` module in the job and all Events use the same instance. Therefore `global` modules use less memory than `stream` modules.